src/Quartz/Job/DirectoryScanJob.cs

```diff
@@ -57,95 +66,52 @@ public DirectoryScanJob()
-            }
-
-            string dirName = mergedJobDataMap.GetString(DirectoryName);
-            string listenerName = mergedJobDataMap.GetString(DirectoryScanListenerName);
-
-            if (dirName == null)
-            {
-                throw new JobExecutionException("Required parameter '" +
-                                                DirectoryName + "' not found in merged JobDataMap");
-            }
-            if (listenerName == null)
-            {
-                throw new JobExecutionException("Required parameter '" +
-                                                DirectoryScanListenerName + "' not found in merged JobDataMap");
-            }
-
-            object temp;
-            schedCtxt.TryGetValue(listenerName, out temp);
-            IDirectoryScanListener listener = (IDirectoryScanListener) temp;
-
-            if (listener == null)
-            {
-                throw new JobExecutionException("DirectoryScanListener named '" +
-                                                listenerName + "' not found in SchedulerContext");
-            }
+            DirectoryScanJobModel model = DirectoryScanJobModel.GetInstance(context);
 
-            DateTime lastDate = DateTime.MinValue;
-            if (mergedJobDataMap.ContainsKey(LastModifiedTime))
+            List<FileInfo> updatedFiles = new List<FileInfo>();
+            Parallel.ForEach(model.DirectoriesToScan, d =>
             {
-                lastDate = mergedJobDataMap.GetDateTime(LastModifiedTime);
-            }
-
-            TimeSpan minAge = TimeSpan.FromSeconds(5);
-            if (mergedJobDataMap.ContainsKey(MinimumUpdateAge))
-            {
-                minAge = TimeSpan.FromMilliseconds(mergedJobDataMap.GetLong(MinimumUpdateAge));
-            }
-
-            DateTime maxAgeDate = DateTime.Now - minAge;
-
-            IEnumerable<FileInfo> updatedFiles = GetUpdatedOrNewFiles(dirName, lastDate, maxAgeDate);
-
-            if (updatedFiles == null)
-            {
-                log.Warn("Directory '" + dirName + "' does not exist.");
-            }
-
-            DateTime latestMod = lastDate;
-            foreach (FileInfo updFile in updatedFiles)
-            {
-                DateTime lm = updFile.LastWriteTime;
-                latestMod = (lm > latestMod) ? lm : latestMod;
-            }
+                var newFiles = GetUpdatedOrNewFiles(d, model.LastModTime, model.MaxAgeDate);
+                if (newFiles != null)
+                {
+                    updatedFiles.AddRange(GetUpdatedOrNewFiles(d, model.LastModTime, model.MaxAgeDate));
+                }
+            });
 
             if (updatedFiles.Any())
             {
                 // notify call back...
-                log.Info("Directory '" + dirName + "' contents updated, notifying listener.");
-                listener.FilesUpdatedOrAdded(updatedFiles);
+                updatedFiles.Select(x => x.DirectoryName)
+                    .Distinct().ToList()
```

Comment:
Maybe just a foreach here for clarity, should already be distinct before this as they are provided to model as is (logged differently otherwise)?